tests/framework/integration/cluster.go

```diff
@@ -179,6 +179,8 @@ type Cluster struct {
 
 	mu            sync.Mutex
 	clusterClient *clientv3.Client
+
+	clusterAuthClients []*clientv3.Client
```

Comment:
There is already a lot of different types of clients stored just to call `.Close()`, maybe we should just store list destructor functions? Even better would be to use `t.Cleanup(func())` for call client.Close.

Reply:
> maybe we should just store list destructor functions

I think that's a good idea. Maybe we could optimize it in a follow-up PR?

Reply:
I mean this PR could just avoid introducing `clusterAuthClients []*clientv3.Client`, by using `cleanup []func()`. We can migrate else in future.

Reply:
okay. Make sense.